rust/datafusion/src/sql/planner.rs

```diff
@@ -515,27 +515,29 @@ impl<S: SchemaProvider> SqlToRel<S> {
                     }
                     _ => match self.schema_provider.get_function_meta(&name) {
                         Some(fm) => {
-                            let rex_args = function
```

Comment:
I wonder given the emphasis on pluggable planers, if 52218c852b7b3016afeaf95d8a46d6deea89d231 (removing the type coercion from physical planner) is a good idea. As in "is it ok to assume that all plans went through the existing `Optimizer` passes before being converted"

It seems reasonable to me, but it might be worth mentioning somewhere (e.g. on the physical planner, etc)

Reply:
AFAI recall, optimizations happen on the logical plan alone. Also, I think that 52218c8 removes the coercion in constructing the logical, not physical, plan (SQL nodes -> physical).

If this is the case, `logical -> physical` does not have to worry about these, right?